java/integTest/src/test/java/glide/cluster/CommandTests.java

```diff
@@ -702,16 +732,37 @@ public void functionLoad(boolean withRoute) {
                         ? clusterClient.functionLoadWithReplace(code, route)
                         : clusterClient.functionLoadWithReplace(code);
         assertEquals(libName, promise2.get());
+        String newFuncName = "myfunc2c_" + withRoute;
         String newCode =
                 code
-                        + "\n redis.register_function('myfunc2c"
-                        + withRoute
+                        + "\n redis.register_function('"
+                        + newFuncName
                         + "', function(keys, args) return #args end)";
         promise2 =
                 withRoute
                         ? clusterClient.functionLoadWithReplace(newCode, route)
                         : clusterClient.functionLoadWithReplace(newCode);
         assertEquals(libName, promise2.get());
+
+        flist =
+                (withRoute
+                                ? clusterClient.functionList(libName, route)
+                                : clusterClient.functionList(libName))
+                        .get();
+        expectedDescription.put(newFuncName, null);
+        expectedFlags.put(newFuncName, Set.of());
+        checkFunctionListResponse(flist, libName, expectedDescription, expectedFlags, Optional.empty());
+
+        flist =
+                (withRoute
+                                ? clusterClient.functionListWithCode(libName, route)
+                                : clusterClient.functionListWithCode(libName))
+                        .get();
+        checkFunctionListResponse(
+                flist, libName, expectedDescription, expectedFlags, Optional.of(newCode));
+
         // TODO test with FCALL
+
+        // TODO FUNCTION FLUSH at the end
```

Comment:
are these just reserving space for commands to be done later?

Reply:
Yes, I removed this TODO in the next PR